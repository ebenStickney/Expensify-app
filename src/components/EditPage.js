import React from 'react';

const EditPage = () => (
<div>
This is my edit page    
</div>
);

export default EditPage; 

