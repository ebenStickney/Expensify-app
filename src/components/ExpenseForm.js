import React from 'react'; 


export default class ExpenseForm extends React.Component {
    state = {
        description: '',
        note: '', 
        amount: ''
    };
    onDescriptionChange = (e) => {
      const description = e.target.value;
      this.setState(() => ({ description }));
    };
    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState( () => ({ note }));
    };
    
    onAmountChange = (e) => {
        const amount = e.target.value;
        
        if (amount.match(/^\d*(\.\d{0,2})?$/)) {
            this.setState( () => ({amount}) );
        }
    };

    render() {
        return (
          <div>
            <h1>Add Expense</h1> 
            <form>
              <input 
                type="text"
                placeholder="Description"
                autoFocus
                value={this.state.description}
                onChange={this.onDescriptionChange}
              />
              <input 
                type='number'
                placeholder='Amount'
                value={this.state.amount}
                onChange={this.onAmountChange}
              />
              <textarea
                placeholder="Add a note (optional)"
                value={this.state.note}
                onChange={this.onNoteChange}
              >
              </textarea>
              <button>Create Expense</button>
            </form>
          </div>
        )
    }
};