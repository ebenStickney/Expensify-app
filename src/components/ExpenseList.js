import React from 'react';
import { connect } from 'react-redux';

const ExpenseList = (props) => (

<div>
 <h4> Expenses </h4>
 {props.name}
</div>

);

const mapStateToProps = (state) => {
    return {
        name: 'Eben'
    }
};

export default connect(mapStateToProps)(ExpenseList);

