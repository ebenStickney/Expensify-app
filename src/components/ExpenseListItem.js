import React from 'react';
import { Link } from 'react-router-dom';


const ExpenseListItem = ({dispatch, id , description, amount, createdAt} ) => (
    <div>
     <Link to={`/edit/${ id }`}> {description} </Link>
     <p>Amount: {amount} -- Created at: {createdAt}</p>
     
    </div>

);

export default ExpenseListItem;