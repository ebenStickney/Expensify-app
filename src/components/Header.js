import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
  <header>
   <h1>Expensify</h1>
   <h4>Manage Your Future</h4>
    <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
    <NavLink to="/create" activeClassName="is-active">Create A New Expense</NavLink>
    <NavLink to="/help" activeClassName="is-active">FAQ</NavLink>
    
  
  </header>
)

export default Header;
