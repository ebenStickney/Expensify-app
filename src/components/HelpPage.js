import React from 'react';

const HelpPage = () => (
<div>
This is where you go for help   
</div>
);

export default HelpPage; 
