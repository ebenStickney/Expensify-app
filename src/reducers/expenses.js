const expenseReducerDefaultState = [];

export default (state = expenseReducerDefaultState, action) => {
    switch (action.type) {
        case "ADD_EXPENSE": 
        return [
            ...state,
            action.expenses
        ];
        break; 
//        
        case "REMOVE_EXPENSE": 
        return state.filter(( {id} ) => action.id !== id);
        break;
//        
        case "EDIT_EXPENSE": 
        return state.map((expense) => {
            if (expense.id === action.id) {
              return {
                  ...expenses, 
                  ...action.updates
              }  
            } else {
                return expenses;
            }
        });
        break; 
            
        default: 
        return state;
    }
};