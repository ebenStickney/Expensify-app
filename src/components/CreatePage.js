import React from 'react';
import { connect } from 'react-redux';
import { startAddExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';


export class CreateExpense extends React.Component {
    onSubmit= (expense) => {
         this.props.startAddExpense(expense);
         this.props.history.push('/');
    };

    render() {
        return (
            <div>
            <ExpenseForm
            onSubmit={this.onSubmit}
            />
            </div>
                );

    }};


const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(CreateExpense);
