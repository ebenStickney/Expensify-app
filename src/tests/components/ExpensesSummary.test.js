import React from 'react';
import { shallow } from 'enzyme'; 
import expenses from '../fixtures/expenses';
import  {ExpensesSummary}  from '../../components/ExpensesSummary';
import selectExpensesTotal from '../../selectors/expenses-total'


test('should render ExpensesSummary Component', () => {
    const expenseCount = expenses.length; 
    const expensesTotal = selectExpensesTotal(expenses); 
    const wrapper = shallow(<ExpensesSummary expenseCount={expenseCount} expensesTotal={expensesTotal} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary Component with single expense', () => {
    const expenseCount = [expenses[1]].length; 
    const expensesTotal = selectExpensesTotal([expenses[1]]); 
    const wrapper = shallow(<ExpensesSummary expenseCount={expenseCount} expensesTotal={expensesTotal} />);
    expect(wrapper).toMatchSnapshot();
});
