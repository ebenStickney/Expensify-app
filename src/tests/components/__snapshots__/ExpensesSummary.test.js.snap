// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary Component 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
       Viewing 
      <span>
        3
      </span>
       
      expenses
       totaling: 
      <span>
        $1,484.07
      </span>
       
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
         Add New Expense 
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpensesSummary Component with single expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
       Viewing 
      <span>
        1
      </span>
       
      expense
       totaling: 
      <span>
        $12.34
      </span>
       
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
         Add New Expense 
      </Link>
    </div>
  </div>
</div>
`;
